nft-rfc-006.md

```diff
@@ -0,0 +1,860 @@
+disclosure of proprietary or personal information problematic.
+
+Second, avoid unnecessary public correlation. DID Documents provide the
+option of embedding service endpoints, which may themselves be present
+in multiple DID Documents. For example, you \*could\* add a Twitter
+address as a service endpoint in several different DIDs. This would link
+those DIDs to each other in a non-verifiable yet undisputable way: while
+observers could not directly tell if those service endpoints mean that
+the Twitter account controls those DIDs, they do gain a point of
+observation for consideration in evaluating that correlation. For many,
+this undermines the privacy goals of being able to use DIDs without
+revealing any additional personally identifiable information.
+
+Third, DIDs, and their sister specification VCs, embrace an open world
+data model, in which anyone can add any parameter to the Document
+without violating conformance. This is convenient, but also means that
+unfortunate properties can and will be used, such as properties that
+explicitly communicate details about the DID Subject. Such details are
+intended to be preserved by Methods and requesting parties even when
+they are misunderstood, which creates significant opportunity for
+unintended consequences. However, it is understood that this flexibility
+affords significant innovation even as it invites endless debate on
+which properties get to be defined where---a decision that has
+significant impact on ubiquitous adoption.
+
+Finally, DIDs are NOT signed documents. The only verification afforded
+by DIDs is that by executing the process defined in the DID Method,
+whatever the result is \*is\* the verified DID Document. This means that
+you MUST trust the software that performs this resolution, either by
+reputation or because you wrote it yourself. This choice was informed by
+a tricky catch-22 where DID identifiers are a result of cryptographic
+operations perform either on or with the DID Document. For example, IPFS
+based DID Documents cannot contain the IPFS address because the content
+hash of that DID Document is the address. Similarly, ledger-based DIDs
+that depend on blockheight determine their ID through the process of
+registration. In both cases, you can\'t fully create the DID Document
+before it is registered. I may be possible to ensure a signature in the
+DID resolver metadata as a signed hash using the cryptographic material
+in that document, but this remains an area of innovation.
+
+**A note on DIDs as URIs rather than IRIs**
+
+DIDs and IIDs are not intended to be human readable, so the spoofing
+concern is mitigated to some degree as is the need to support
+alternative character sets. In fact, DIDs and IIDs are generally
+encodings of cryptographic identifiers, such as a public key; none of
+the extant encodings use non-ascii characters.
+
+Proposal 
+---------
+
+**First**, create IIDs as an interoperable family DIDs with additional
```

Comment:
Should we define if DID's are extensible, then if they are not, should create IID's? :thinking: 

https://www.w3.org/2019/08/did-20190828/#extensibility

Reply:
Yes. IIDs are DIDs and we should clarify the extensibility mechanism. This wanders into some interesting territory about whether or not IIDs should support both JSON and JSON-LD representations. If we do, we would need to add any IID properties to the DID Spec Registry *and* create an IID context that is specified in IID Documents. It would also probably be useful to have a registry of IID specific extensions for reference.

One of my main advocacies with regard to IIDs is that IID Subjects are always on-chain assets, which then establish their own namespace for resources and references. Resources are downloadable and use the /path part of the DID URL. References are pure URIs, and use the #fragment part. Currently, this is not a pattern advocated for DID Documents, as they are anticipated for use with ANY subject.

Although all IIDs are DIDs, not all DIDs are going to be IIDs.

That also gives us an opportunity to specify IID properties (in the IID Document) that are uniquely suited to on-chain assets like tokens. For those who want interoperable features for on-chain assets, IIDs would be more likely to meet their need, whatever chain they are on, precisely because they could be expected to use IID properties.

Reply:
Thanks for explaining.

> IID Subjects are always on-chain assets

What is an on-chain vs off-chain asset? 

on-chain = NFT to represent shares of an asset 
off-chain = the asset itself

:thinking:

Reply:
On-chain means that it is an asset that resides, in some fashion, on the decentralized substrate of a ledger. That could be an NFT, an address, a smart contract, a UTXO, etc.

In this proposal, the base IID, e.g., 
```
did:ixo:abc123
```
**always** refers to an on-chain asset (in this case an NFT representing a CER cert issued under the authority of the UN Climate Change).

To refer to other assets, which may or may not be on-chain, an IID defines its own namespace with #fragment parts, e.g.,
```
did:ixo:abc123#vc1
```
Refers to a verifiable credential associated with the NFT. This URI can be used for assertions about the VC.

Similarly,
```
did:ixo:abc123#parcel1
```
Refers to a real-world parcel of land, which could be described with additional statements about that parcel, such as its geographic boundaries.

IID References are free to refer to any Subject.

In contrast, to specify a retrievable resource, you use a /path part:
```
did:ixo:abc123/vc1
```
This means, look up the linkedResource with that path, and return the associated resource.

In this usage, you could, for example, present this vc (as raw json-ld) in an iframe with a tag like this:
```
<iframe src="did:ixo:abc123/vc1"></iframe>
```

The metadata needed to download that resource is specified as an entry in the LinkedResources property.

It's this convention that helps us avoid HTTPRange14 issues:
1. All IID Subjects are on-chain assets
2. IID References are URIs (namespace defined by the on-chain asset) suitable for RDF assets
3. IID Resources are retrievable resources specified in LinkedResources, usable anywhere a URL might be used
4. For IID Resources, define a matching IID Reference, so all downloadable resources also have a clean URI for assertions. Use the IID Reference as a URI rather than use the IID Resource as a URI.

So, the IID Resource ```did:ixo:abc123/vc1``` has a matching IID Reference ```did:ixo:abc123#vc1```. 
If we construct a similar iframe element as above, we can see the difference between these two.
```
<iframe src="did:ixo:abc123#vc1"></iframe>
```
This iframe, using the fragment pattern for IID References, would display the #vc1 entry in the IID Document, which is the part of the JSON-LD document specifying the linkedResource, e.g., the second iframe element would show something like this:
> {
“id”: “did:ixo:abc123#vc1”,
“path”: "did:ixo:abc123/vc1,
“type” : “verifiableCredential”, // note we need the context from vcs
“proofType” : “multihash” ,
“proof” : “QmYtUc4iTCbbfVSDNKvtQqrfyezPPnFvE33wFmutw9PBBk”,
“endpoint” : “https://producer.example.com/abc123/vc1.json”
}

Recall that the previous iframe displayed the actual VC and not the metadata about where to retrieve it.

This is how we can have clean identifiers that avoid the HTTPRange14 problem, through an adopted convention of the IID family.